File: market-mapper/src/main/java/by/market/mapper/EntityMetadataMapper.java
package by.market.mapper;


import by.market.domain.system.EntityMetadata;
import by.market.dto.system.EntityMetadataDTO;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(config = MapperConfiguration.class, uses = ContainerMetadataMapper.class)
public interface EntityMetadataMapper extends MapstructMapper<EntityMetadataDTO, EntityMetadata> {

    @Override
    @Mapping(source = "container.id", target = "container")
    EntityMetadataDTO toMap(EntityMetadata entityMetadata);

    @Override
    @Mapping(source = "container", target = "container.id")
    EntityMetadata fromMap(EntityMetadataDTO entityMetadataDTO);

}

File: market-mapper/src/main/kotlin/by/market/mapper/system/EntityMetadataMapper.kt
package by.market.mapper.system

import by.market.domain.system.EntityMetadata
import by.market.dto.system.EntityMetadataDTO
import by.market.mapper.IMapstructMapper
import by.market.mapper.MapperConfig
import org.mapstruct.Mapper

@Mapper(config = MapperConfig::class)
interface EntityMetadataMapper : IMapstructMapper<EntityMetadataDTO, EntityMetadata>
